app/static/audio_settings.js

```diff
@@ -0,0 +1,174 @@
+  if (elem) {
+    elem.addEventListener("click", func);
+  }
+}
+
+/**
+ * @summary This function gets an element_id and set its checked attribute.
+ * According to the element_id, we disable or enable track selection and volume change for that audio type.
+ * @param {string} element_id - the id attribute of the html element. One of: Music On/off, Sound Effects On/Off.
+ */
+function set_checked(element_id) {
+  const is_music = element_id.includes("music");
+  const is_off = element_id.includes("off");
+  const to_toggle = is_music ? "music" : "sfx";
+  set_disabled_or_enabled(to_toggle, is_music, is_off);
+}
+
+/**
+ * @summary This function sets audio options off by default.
+ */
+function set_default() {
+  set_default_for_audio_type("music-on", "music-off");
+  set_default_for_audio_type("sound-effects-on", "sound-effects-off");
+}
+
+/**
+ * @summary This function gets class or id name, boolean value to tell if class or id, and bolean value to set,
+ * And sets the disabled attribute of the corresponding element accordingly.
+ * @param {string} name - name of the elements' class or id.
+ * @param {Boolean} is_class - class if true, id otherwise.
+ * @param {Boolean} to_set - we set the disabled attribute if true, false oterwise.
+ */
+function set_disabled_or_enabled(name, is_class, to_set) {
+  if (is_class) {
+    let elements = document.getElementsByClassName(name);
+    for (let element of elements) {
+      element.disabled = to_set;
+    }
+  } else {
+    document.getElementById(name).disabled = to_set;
+  }
+  document.getElementById("rangeInput-" + name).disabled = to_set;
+}
+
+/**
+ * @summary This function is an helper function for the set_default function,
+ * and we use it to privent code duplication by having one function to handle music as well as sound effects.
+ * @param {string} audio_id_on - the id corresponding to the On option of the element, for Music as well as sfx.
+ * @param {string} audio_id_off - the id corresponding to the Off option of the element, for Music as well as sfx.
+ */
+function set_default_for_audio_type(audio_id_on, audio_id_off) {
+  const is_on = document.getElementById(audio_id_on).checked;
+  const is_off = document.getElementById(audio_id_off).checked;
+  if (!is_on && !is_off) {
+    document.getElementById(audio_id_off).checked = true;
+  }
+}
+
+function prepare_audio() {
+  let audio_settings = JSON.parse(this.response);
+  const music = document.getElementById("my-audio");
+  const sfx = document.getElementById("sfx");
+  audio_settings = JSON.parse(audio_settings);
+  const music_on = audio_settings["music_on"];
+
+  if (music.muted && (music_on || music_on == null)) {
+    const choices = audio_settings["playlist"];
+    music.src = `/static/tracks/${
+      choices[Math.floor(Math.random() * choices.length)]
+    }`;
+    music.volume = audio_settings["music_vol"];
+    music.muted = false;
+  }
+
+  if (music.paused) {
+    music.play();
+  }
+
+  sfx_on = audio_settings["sfx_on"];
```

Comment:
Please insert the JS to ESLint and lint it. That's not the first time of undeclared variables here :)

Reply:
Sorry, and thanks for the tip! It works great.